scene/gui/spin_box.cpp

```diff
@@ -114,8 +121,15 @@ void SpinBox::_line_edit_input(const Ref<InputEvent> &p_event) {
 void SpinBox::_range_click_timeout() {
 	if (!drag.enabled && Input::get_singleton()->is_mouse_button_pressed(MouseButton::LEFT)) {
 		bool up = get_local_mouse_position().y < (get_size().height / 2);
-		double step = get_custom_arrow_step() != 0.0 ? get_custom_arrow_step() : get_step();
-		set_value(get_value() + (up ? step : -step));
+		bool original_do_round = is_using_rounded_values();
+		double step = get_step();
+		double temp_step = get_custom_arrow_step() != 0.0 ? get_custom_arrow_step() : get_step();
+		_rounded_values = false;
```

Comment:
Why this? `rounded` should always make the value rounded.

Reply:
Oh sorry, now that you mention it, I think it should be removed too. It's been awhile but I think the reasoning was the value was not updating if the arrow_step was too low. But I guess that logic is flawed. I'll remove it

Reply:
Oh ok, so if i set the step to 0.8 the value cannot get past 2.

Reply:
It's a pre-existing bug, so it can be fixed later.
Not sure what's the expected behavior here, maybe `rounded` could force the arrow step to be at least 1.
Then again, when both steps are 0, the arrows don't work either.